src/dreyfus_util.erl

```diff
@@ -166,6 +168,71 @@ time(Metric, {M, F, A}) when is_list(Metric) ->
         couch_stats:update_histogram([dreyfus | Metric],  Length)
     end.
 
+in_black_list(DbName, GroupId, IndexName) when is_list(DbName),
+        is_list(GroupId), is_list(IndexName) ->
+    in_black_list([DbName, GroupId, IndexName]);
+in_black_list(DbName, GroupId, IndexName) when is_binary(DbName),
+        is_binary(GroupId), is_binary(IndexName) ->
+    in_black_list([?b2l(DbName), ?b2l(GroupId), ?b2l(IndexName)]);
+in_black_list(_DbName, _GroupId, _IndexName) ->
+    false.
+
+in_black_list(IndexEntry) when is_list(IndexEntry) ->
+    BlackList = dreyfus_config:get(black_list),
+    lists:member(IndexEntry, BlackList);
+in_black_list(_IndexEntry) ->
+    false.
+
+maybe_deny_index(DbName, GroupId, IndexName) ->
+    case in_black_list(DbName, GroupId, IndexName) of
+        true ->
+            Reason = ?l2b(io_lib:format("Index <~s, ~s, ~s>, is BlackListed",
+                [?b2l(DbName), ?b2l(GroupId), ?b2l(IndexName)])),
+            throw ({bad_request, Reason});
+        _ ->
+            ok
+    end.
+
+add_bl_element(DbName, GroupId, IndexName) when is_list(DbName),
```

Comment:
I didn't see that `add_bl_element/3` and `add_bl_element/1` are called in production codes (they are called in test code for sure). I can understand that they can be used for operator to add entry to blacklist. Just wonder whether there is need to have meaning response to operators and let them know whether entry is added or not?

Reply:
@jiangphcn : we need to to add or remove from the list. otherwise, the operator has to call ```config:set/3``` for an entire new list. Remember, config doesn't have adding or removing from list capabilities.

Reply:
So `add_bl_element/3` will be called by operator to add or remove entity from list, right? My comment is whether we need to have meaningful return code/message to operator so that they know the actual result.

Reply:
@jiangphcn : I understand your idea. Perhaps we can move this over to snippet with a more meaningful response instead since it is not used in the source code?